Persist visit options if a response is present

olivaresf opened the pull request:
Persist visit options for when a response HTML is provided. I've seen this fix an issue where the flash alert would not be displayed, if provided via visit options.

olivaresf commented on Source/Session/VisitOptionsHandler.swift:
```diff
@@ -0,0 +1,24 @@
+            return options
+        } else if let unhandledVisitOptions {
+            /// Next visit is happening. Use the previous visit options.
+            self.unhandledVisitOptions = nil
```
Make sure URLs match between visit options

joemasilotti replied:
Has this been resolved?

olivaresf replied:
Not exactly. Would we keep the visit options around indefinitely until the next visit or should the next visit use them? If the former, URL matching makes sense. Otherwise, this implementation works for our purposes.

What do you think? Sorry to say I'm still trying to understand when to expect a `responseHTML` and I also don't think Android is doing it 🤔.

seanpdoyle commented:
@olivaresf is this work related to https://github.com/hotwired/turbo-ios/issues/136#issuecomment-1944578518? What kind of community support could help push this forward?

olivaresf commented:
Coincidentally, my plan is to test this PR thoroughly sometime next week. Let me test this and https://github.com/hotwired/turbo-ios/pull/160 as I believe we're both sort of circling around the same issue and then I'll come back to you.

joemasilotti commented:
Hey @olivaresf, we'd love to get this wrapped up as it also fixes #186.

Is all that's left adding tests and addressing the comment?

I'm no Android expert, but @jayohms called out that the Android library handles this in a separate view model to [keep track of the visit options](https://github.com/hotwired/turbo-android/blob/main/turbo/src/main/kotlin/dev/hotwire/turbo/nav/TurboNavigator.kt#L227-L230) in between requests. It's then used [here](https://github.com/hotwired/turbo-android/blob/main/turbo/src/main/kotlin/dev/hotwire/turbo/delegates/TurboWebFragmentDelegate.kt#L252-L255).

I wonder if there's an approach that might work better on iOS following that model? If you've already explored that then disregard, just wanted to explore a closer implementation between the two libraries.

olivaresf commented:
Thanks for your patience! Still trying to come back to this, but I've had other priorities.

> Is all that's left adding tests and addressing the comment?

Yep. If you can help me with that, I'd appreciate it. If not, I'll make some time tomorrow to test this out.

olivaresf commented:
I've added a unit test for this behavior. As far as I can tell, I'm mimicking Android's behavior:

- If visit options with a `responseHTML` are provided during a Turbo visit, we save the options.
- During the next visit, saved options are given to the new visit and then saved options are deleted so they cannot be reused.

I've tested this on Basecamp and it works as I expected. Let me know if there's anything else needed.

jayohms reviewed (approved):
This looks good to me 👍. I looked at the Android side and we're not validating the url between visits for the response, so in practice I don't see this as an issue.
